web/src/recent_view_ui.ts

```diff
@@ -1196,19 +1191,23 @@ function recenter_focus_if_off_screen(): void {
     }
 }
 
-function is_scroll_position_for_render(scroll_container: HTMLElement): boolean {
-    const table_bottom_margin = 100; // Extra margin at the bottom of table.
-    const table_row_height = 50;
-    return (
-        scroll_container.scrollTop +
-            scroll_container.clientHeight +
-            table_bottom_margin +
-            table_row_height >
-        scroll_container.scrollHeight
-    );
+function is_scroll_position_for_render(): boolean {
+    // We render new rows if user has already scrolled 2 / 3 of rendered row.
+    const scroll_position = window.scrollY;
+    const scroll_height = window.innerHeight;
+    const compose_max_height = $("html").css("--max-unexpanded-compose-height");
+    assert(typeof compose_max_height === "string");
+    const scroll_max = document.body.scrollHeight - Number.parseInt(compose_max_height, 10);
+    return scroll_position + scroll_height >= (2 / 3) * scroll_max;
 }
```

Comment:
Can you tweak this function to document why  `--max-unexpanded-compose-height` is the right compose height to use?

Reply:
Added this comment:


```
    // `--max-unexpanded-compose-height` is just empty space below the last rendered row
    // in recent view. We don't want user to see this empty space until there are no new
    // rows to render when the user is scrolling to the bottom of the view. So, we render new
    // rows when user has scrolled 2 / 3 of (the total scrollable height - the empty space).
```